packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -347,6 +348,36 @@ Future<void> buildGradleProject({
   }
 }
 
+// TODO(mklim): Fix the obvious bugs.
+// - Do we need logic for passing along build architecture, local engine, etc?
```

Comment:
Yes, we really should support that.

That said, I'm not sure the plugins actually need libflutter.so - while it's probably possible for some plugin to try to use JNI on that, I think it's probably very uncommon and we're safer to break that (totally a gut sense though, I have no hard data).  This is where it'd be nice to just have a set of interfaces the engine exposes in a lighter weight JAR that plugins could use.